glusterd2/commands/volumes/brick-replace.go

```diff
@@ -0,0 +1,295 @@
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Get source brick information like size etc
+	brickInfo, err := volume.BrickStatus(srcBrickInfo, mtabEntries)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Get new brick from the available vgs
+	newBrick := getNewBrick(availableVgs, brickInfo, vol, brickCount)
+
+	peerID := uuid.Parse(newBrick.PeerID)
+	if peerID == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "peer id of new brick could not be parsed")
+		return
+	}
+
+	nodes := []uuid.UUID{peerID}
+	txn, err := transaction.NewTxnWithLocks(ctx, volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+	defer txn.Done()
+
+	txn.Steps = []*transaction.Step{
+		{
+			DoFunc: "brick-replace.txnPrepareBrick",
+			Nodes:  nodes,
+		},
+		{
+			DoFunc: "brick-replace.ReplaceVolinfo",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		{
+			DoFunc:   "vol-create.InitBricks",
+			UndoFunc: "vol-create.UndoInitBricks",
+			Nodes:    nodes,
+		},
+		{
+			DoFunc: "brick-replace.StartBrick",
+			Nodes:  nodes,
+		},
+		{
+			DoFunc:   "vol-create.StoreVolume",
+			UndoFunc: "vol-create.UndoStoreVolume",
+			Nodes:    []uuid.UUID{gdctx.MyUUID},
+		},
+	}
+	if err := txn.Ctx.Set("newBrick", &newBrick); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+	if err := txn.Ctx.Set("srcBrickInfo", &srcBrickInfo); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+	if err := txn.Ctx.Set("subVolIndex", &subVolIndex); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+	if err := txn.Ctx.Set("brickIndex", &brickIndex); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+	if err := txn.Ctx.Set("volinfo", &vol); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+	if err := txn.Do(); err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+
+	return
+}
+
+func getNewBrick(availableVgs []bricksplanner.Vg, brickInfo brick.Brickstatus, vol *volume.Volinfo, brickCount int) api.BrickReq {
+	var newBrick api.BrickReq
+	for _, vg := range availableVgs {
+		if vg.AvailableSize >= brickInfo.Size.Capacity {
+			brickSize := brickInfo.Size.Capacity
+			lvName := fmt.Sprintf("brick_%s_s%d_b%d", vol.Name, brickCount, brickCount)
+			brickTpSize := uint64(float64(brickSize) * vol.SnapshotReserveFactor)
+			newBrick = api.BrickReq{
+				Type:           "Brick",
+				Path:           brickInfo.Info.Path,
+				Mountdir:       "/brick",
+				TpName:         fmt.Sprintf("tp_%s_s%d_b%d", vol.Name, brickCount, brickCount),
+				LvName:         lvName,
```

Comment:
@rishubhjain  Shouldn't LvName and TpName contain subvolIndex and brickIndex instead of brickCount.